pkg/reconciler/configmap/configmap.go

```diff
@@ -0,0 +1,396 @@
+	cache.Spec.Template.Spec.Containers[0].Env = []corev1.EnvVar{
+		{Name: "CACHE_PATH", Value: "/cache"},
+		{Name: "QUARKUS_VERTX_EVENT_LOOPS_POOL_SIZE", Value: "4"},
+		{Name: "QUARKUS_THREAD_POOL_MAX_THREADS", Value: "50"},
+	}
+	//central is at the hard coded 200 position
+	repos := []Repo{{name: "central", position: 200}}
+	if rebuildsEnabled {
+		repos = append(repos, Repo{name: "rebuilt", position: 100})
+		cache.Spec.Template.Spec.Containers[0].Env = append(cache.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  "QUARKUS_S3_ENDPOINT_OVERRIDE",
+			Value: "http://" + LocalstackDeploymentName + "." + request.Namespace + ".svc.cluster.local:4572",
+		})
+		cache.Spec.Template.Spec.Containers[0].Env = append(cache.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  "QUARKUS_S3_AWS_REGION",
+			Value: "us-east-1",
+		})
+		cache.Spec.Template.Spec.Containers[0].Env = append(cache.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  "QUARKUS_S3_AWS_CREDENTIALS_TYPE",
+			Value: "static",
+		})
+		cache.Spec.Template.Spec.Containers[0].Env = append(cache.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  "QUARKUS_S3_AWS_CREDENTIALS_STATIC_PROVIDER_ACCESS_KEY_ID",
+			Value: "accesskey",
+		})
+		cache.Spec.Template.Spec.Containers[0].Env = append(cache.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  "QUARKUS_S3_AWS_CREDENTIALS_STATIC_PROVIDER_SECRET_ACCESS_KEY",
+			Value: "secretkey",
+		})
+	}
+	regex, err := regexp.Compile("maven-repository-(\\d+)-(\\w+)")
+	for k, v := range configMap.Data {
+		if regex.MatchString(k) {
+			results := regex.FindStringSubmatch(k)
+			atoi, err := strconv.Atoi(results[1])
+			name := results[2]
+			if err != nil {
+				return err
+			}
+			cache.Spec.Template.Spec.Containers[0].Env = append(cache.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+				Name:  "STORE_" + strings.ToUpper(name) + "_URL",
+				Value: v,
+			})
+			cache.Spec.Template.Spec.Containers[0].Env = append(cache.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+				Name:  "STORE_" + strings.ToUpper(name) + "_TYPE",
+				Value: "maven2",
+			})
+			repos = append(repos, Repo{position: atoi, name: name})
+		}
+	}
+
+	var sb strings.Builder
+	sort.Slice(repos, func(i, j int) bool {
+		return repos[i].position < repos[j].position
+	})
+	for _, i := range repos {
+		if sb.Len() > 0 {
+			sb.WriteString(",")
+		}
+		sb.WriteString(i.name)
+	}
+	cache.Spec.Template.Spec.Containers[0].Env = append(cache.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+		Name:  "BUILD_POLICY_DEFAULT_STORE_LIST",
+		Value: sb.String(),
+	})
+	cache.Spec.Template.Spec.Containers[0].Image = r.configuredCacheImage
+
+	role := v1alpha1.Role{}
+	roleName := types.NamespacedName{Namespace: request.Namespace, Name: "pipeline-anyuid-role"}
+	err = r.client.Get(ctx, roleName, &role)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			role := v1alpha1.Role{}
+			role.Name = "pipeline-anyuid-role"
```

Comment:
Preface: I understand the need for this change here with what this PR is accomplishing.

That said, a controller adding roles and rolebindings (i.e. manipulate permissions) is atypical (and can in fact be frowned upon).

Now, needing anyuid is a function of the base images we are leveraging during our builds not giving us write access to content we need.

We are not the only item under the appstudio umbrella fighting this (the buildah task is another one, where it has a path around user workspaces for eventually solving this (though that is not there in openshift 4.10 IIRC so that I think is a big change).

But I have had brief conversations with Adam Kaplan about "long term" not wanting any appstudio components to leverage anyuid.

Bottom line, *minimally* we need a big TODO comment here about we know we need to see about create images that do not need anyuid, where that is complex enough that if we do not already have a Jira story that tracks this work, we should either update an existing story, or create a new one.

@Michkov FYI

Reply:
This PR actually changes the base image to the ones built in openshift, so I can probably make that image change as part of this PR. 

I will need to do some local testing.

Reply:
ok cool - we'll play it by ear.

initial thought looking for example (but not limited to) we would tweak https://github.com/redhat-appstudio/jvm-build-service/blob/main/Dockerfile.sidecar-in-ci so that before we set the user to non-root at https://github.com/redhat-appstudio/jvm-build-service/blob/main/Dockerfile.sidecar-in-ci#L18 we create dirs and update permissions so that we no longer the various `chown` calls and use of anyuid scc in our various tasks (i.e. https://github.com/redhat-appstudio/jvm-build-service/blob/3a19938e522f16830cb6c67449ad0bf5117ffc3c/pkg/reconciler/dependencybuild/buildrecipeyaml.go#L253-L254)

The only task not unique to jvm build service is the git-clone task (which is in build-serivce, but I think is basically a copy of the tekton cluster task.  I think it would need to be solved as well long term, but since that effects everyone, probably would need a broader discussion / review.